pytools/test/test_persistent_dict.py

```diff
@@ -899,8 +899,109 @@ def method(self):
     # }}}
 
 
+# {{{ mpi test infrastructure
+
+def run_test_with_mpi(num_ranks, f, *args, extra_env_vars=None):
```

Comment:
MPI seems like a heavy-handed thing to use here. Wouldn't [multiprocessing](https://docs.python.org/3/library/multiprocessing.html) suffice?

Reply:
What do you think of https://github.com/inducer/pytools/pull/231/commits/cc9966cd155dd5e92d09063cb879ce617f331ffe? (I haven't worked with `multiprocessing` before, so I'm not sure if that code is idiomatic). The test still fails in #232 without the concurrency improvements here.